adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -90,6 +92,19 @@ def gethostbyname(hostname):
     return addr
 
 
+def is_ipv4(host):
+    """Checks if a host string is an IPv4 address.
+    :param str host: host's name or ip
+    """
+    octets = host.split(".", 3)
```

Comment:
Users provide the IP address, please use `rstrip`[1] on L99 to reduce the chance of L100 tossing errors: `host.rstrip().split(".", 3)` 

[1] https://python-reference.readthedocs.io/en/latest/docs/str/rstrip.html)

Reply:
Similar to the comment above, the `isdigit()` check will ensure that any extra special, white space, or alpha characters will cause a `False` return. If I were to `rstrip()` here, the string "1.2.3.4 " (note the last character is a space) would return `True` indicating that it is an IPv4 address. This string, space included, would then be returned by `getaddrinfo()` causing issues or additional checks to be necessary further down the stack. 

Now, I admit, this _will_ cause a DNS lookup to be performed for the malformed IP address. But without additional checks on what constitutes a valid hostname/domain (which would be very out of place here), a `NXDOMAIN` is ultimately the most reliable means of validating the data and altering the user to their error.